doc/manual/src/command-ref/nix-build.md

```diff
@@ -12,6 +12,11 @@
   [`--dry-run`]
   [{`--out-link` | `-o`} *outlink*]
 
+# Disambiguation
+
+This man page describes the command `nix-build`, which is distinct from `nix
+build`. For documentation on the latter, see `man nix3-build`.
```

Comment:
> (afaik --help is the recommended way of getting the help because it's more flexible than the manpage)

`nix build --help` does not respect things like `MANPAGER`, so it's not very convenient. I'd rather be conservative here and shield potentially new users from quirks like these.

In what way is it more flexible?

Reply:
Given that this is a man page, I also think it makes sense to point to the **correct** man page. The main goal is for newcomers to learn where they can find the correct man page and the difference between the two.

Reply:
> Also the manpages rendering is a bit ugly compared to --help (and will probably remain a bit of a second-class citizen)

Quite the opposite in my case, since `--help` ignores `MANPAGER`.

Ugly is a matter of taste, TBH. `man` will use the user's standard layout, `--help` will use the one hand-picked by nix.

I've updated the patch, hopefully this is a good compromise. :)